src/typeChecker/typeErrorChecker.ts

```diff
@@ -1280,20 +1465,20 @@ function substituteVariableTypes(type: Type, typeVar: Variable, typeToSub: Type)
 /**
  * Combines all types provided in the parameters into one, removing duplicate types in the process.
  */
-function mergeTypes(...types: Type[]): Type {
+function mergeTypes(node: tsEs.Node, ...types: Type[]): Type {
   const mergedTypes: Type[] = []
   for (const currType of types) {
     if (isEqual(currType, tAny)) {
       return tAny
     }
     if (currType.kind === 'union') {
       for (const type of currType.types) {
-        if (!containsType(mergedTypes, type)) {
+        if (!containsType(node, mergedTypes, type, [], [], true)) {
```

Comment:
Why do we skip type alias expansion here?

Reply:
`mergeTypes` is called when removing duplicate types from union types, in which case it is sufficient to compare the variable types at name level without expanding them.

Reply:
Hmm, would it be good to leave a comment here explaining why (coming from a long-term maintainability perspective)?